apps/prairielearn/src/pages/instructorInstanceAdminSettings/instructorInstanceAdminSettings.html.ts

```diff
@@ -0,0 +1,266 @@
+                        </button>
+                      </div>
+                    </span>
+                    <div class="d-none">
+                      <div id="js-student-link-qrcode">
+                        <center>${unsafeHtml(studentLinkQRCode)}</center>
+                      </div>
+                    </div>
+                  </td>
+                </tr>
+              </tbody>
+            </table>
+            ${resLocals.authz_data.has_course_permission_edit && !resLocals.course.example_course
+              ? html`
+                  <div class="card-footer">
+                    <div class="row">
+                      ${CopyCourseInstanceForm({
+                        csrfToken: resLocals.__csrf_token,
+                        shortName: resLocals.course_instance.short_name,
+                      })}
+                    </div>
+                  </div>
+                `
+              : ''}
+          </div>
+        </main>
+      </body>
+    </html>
+  `.toString();
+}
+
+function EditConfiguration({
+  hasCoursePermissionView,
+  hasCoursePermissionEdit,
+  exampleCourse,
+  urlPrefix,
+  navPage,
+  infoCourseInstancePath,
+}: {
+  hasCoursePermissionView: boolean;
+  hasCoursePermissionEdit: boolean;
+  exampleCourse: boolean;
+  urlPrefix: string;
+  navPage: string;
+  infoCourseInstancePath: string;
+}) {
+  if (!hasCoursePermissionView && !hasCoursePermissionEdit) {
+    return 'infoCourseInstance.json';
+  }
+
+  if (hasCoursePermissionEdit && !exampleCourse) {
+    return html`
+      <a href="${urlPrefix}/${navPage}/file_view/${infoCourseInstancePath}">
+        infoCourseInstance.json
+      </a>
+      <a
+        class="btn btn-xs btn-secondary mx-2"
+        role="button"
+        href="${urlPrefix}/${navPage}/file_edit/${infoCourseInstancePath}"
+      >
+        <i class="fa fa-edit"></i>
+        <span>Edit</span>
+      </a>
+    `;
+  } else {
+    return html`
+      <a href="${urlPrefix}/${navPage}/file_view/${infoCourseInstancePath}">
+        infoCourseInstance.json
+      </a>
+    `;
+  }
+}
+
+function CopyCourseInstanceForm({
+  csrfToken,
+  shortName,
+}: {
+  csrfToken: string;
+  shortName: string;
+}) {
+  return html`
+    <div class="col-auto">
+      <form name="copy-course-instance-form" class="form-inline" method="POST">
+        <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+        <button name="__action" value="copy_course_instance" class="btn btn-sm btn-primary">
+          <i class="fa fa-clone"></i> Make a copy of this course instance
+        </button>
+      </form>
+    </div>
+    <div class="col-auto">
+      <button
+        class="btn btn-sm btn-primary"
+        href="#"
+        data-toggle="modal"
+        data-target="#deleteCourseInstanceModal"
+      >
+        <i class="fa fa-times" aria-hidden="true"></i> Delete this course instance
+      </button>
+    </div>
+    ${Modal({
+      id: 'delete-course-instance-form',
```

Comment:
It seems like you didn't test this after changing it? This doesn't match the `data-target` used above (`#deleteCourseInstanceModal`) so the modal will never open.

```suggestion
      id: 'deleteCourseInstanceModal',
```

You'll have to adjust the tests too. Note that this ID is not attached to a `<form>`, so we should not include `-form` in the name.

Reply:
Updated this as well as the tests in [b354de3](https://github.com/PrairieLearn/PrairieLearn/pull/9673/commits/b354de348fa4ec5559c6b0a4b48afd1e7ca76e73). 

We had used the other naming syntax (`delete-question-form`) when implementing a similar modal for deleting questions. I will update the ID of that one in subsequent PR to match this.